docs/content/develop/properties.md

```diff
@@ -0,0 +1,348 @@
+
+| Property | Attribute |
+| --- | --- |
+| `4` | `"4"` |
+| `-10` | `"-10"` |
+| `NaN` | `"NaN"` |
+| `Infinity` | `"Infinity"` |
+| `-Infinity` | `"-Infinity"` |
+| `undefined` | _removes attribute_ |
+| `null` | _removes attribute_ |
+
+Examples: `this.count = NaN` (or `Number.NaN`) will result in `pfe-c-count="NaN"`.
+
+Valid values: any Number, `NaN`, `null`, and `undefined`.
+
+### Boolean
+
+Boolean properties behave just like [HTML boolean attributes](https://html.spec.whatwg.org/dev/common-microsyntaxes.html#boolean-attribute) you might be familiar with, like `disabled` or `checked`.  
+
+```javascript
+isLoading: {
+  type: Boolean
+}
+```
+
+| Property | Attribute |
+| --- | --- |
+| `true` | `""` |
+| `false` | _removes attribute_ |
+| `"hello I am true"` | `""` |
+| `undefined` | _removes attribute_ |
+| `null` | _removes attribute_ |
+
+When referencing a Boolean attribute, use `hasAttribute(name)` rather than `getAttribute(name)`.
+
+### Handling `null` and `undefined`
+
+If a property is assigned to `null` or `undefined`, its associated attribute will be removed.  This is true regardless of the property's `type`.
+
+Example: `this.count = null` will result in the `pfe-c-count` attribute being removed.
+
+## default
+
+Default: `undefined`
+
+During the component's `connectedCallback`, values with defined defaults will be initialized in both properties and attributes.
+
+```javascript
+count: {
+  type: Number,
+  default: 0
+}
+```
+
+Example: since `count` has a default of `0`, when the component is connected and wasn't given a `pfe-c-count` attribute already, one will be created with the default value `0`.  In other words, `this.count === 0` and `this.getAttribute("pfe-c-count") === "0"`.
+
+## observer
+
+Default: `undefined`
+
+Observers provide a quick way to wire up a function to be called whenever a value changes.
+
+You may provide an `observer` for any property, which is the string name of a (non-static) function inside the web component which will be called whenever the property changes.  Here's an example wherein changes to `count` will trigger the `handleCount` function.
+
+```javascript
+class PfeCount extends PFElement {
+
+  /* some boilerplate omitted */
+
+  static get properties() {
+    return {
+      count: {
+        type: Number,
+        observer: "handleCount"
+      }
+    }
+  }
+
+  handleCount(oldVal, newVal) {
+    console.log(`count changed from ${oldVal} to ${newVal}`);
+  }
+
+}
+```
+
+Observer functions are called with arguments `(oldVal, newVal)`.  Like property values, both arguments will be cast to the appropriate type, ie String, Number, or Boolean.
+
+
+## cascade
+
+Default: `undefined`
+
+`cascade` allows an attribute value to be automatically copied to one or more child elements.  The value of `cascade` is a CSS-style selector which is used to match the children that should receive the values.  The selector is applied to the element's children in both the light DOM and the shadow DOM.
```

Comment:
Does this accept any selector?  Could I do: `"cascade": "*"` for example?  Or `"cascade": ":not(.foo)"`

Reply:
Asked directly and the answer was yes!  Anything querySelectorAll accepts.